### ISSUE TITLE
Perplexity info bug for new version

### ISSUE THREAD

**Issue by netxiao** [OpenNMT/OpenNMT#161]
**new version training log:**
[03/09/17 13:19:04 INFO] Epoch 9 ; Iteration 134275/134275 ; Learning rate 0.1250 ; Source tokens/s 802 ; Perplexity **12.01**
[03/09/17 13:24:51 INFO]  Validation perplexity: **320.86**
[03/09/17 13:24:51 INFO] Saving checkpoint to 'model_common1_epoch9_320.86.t7'...

**old version:** 
[03/02/17 06:16:06 INFO] Epoch 18 ; Iteration 24547/24547 ; Learning rate 0.1250 ; Source tokens/s 2769 ; Perplexity **11.81**
[03/02/17 06:17:21 INFO] Validation perplexity: **9.75**
[03/02/17 06:17:21 INFO] Saving checkpoint to 'model_it/old_it_epoch18_9.75.t7'...

**Comment by guillaumekln** [comment]
What is the bug exactly and which versions are you comparing? It does not look like the same training.

**Comment by netxiao** [comment]
In previous versions training log, each Iteration of perplexity and validation perplexity were similar(Perplexity 11.81,Validation perplexity: 9.75), but in the latest version,the validation perplexity value make me confused(Perplexity 12.01,Validation perplexity: 320.86).

**my recent epoch model **:
```
model_common1_epoch7_17.91.t7
model_common1_epoch8_17.79.t7
model_common1_epoch9_320.86.t7
model_common1_epoch10_277.03.t7
```

**Comment by guillaumekln** [comment]
Can you be more precise about what you mean by "old" and "new" versions? Also what are your training options?

**Comment by netxiao** [comment]
"old" version: March 7 before the source code
"new" version: March 9 after the source code
The problem appeared when I used `git pull `to update the source code on March 7.

my training options:
```
start_epoch = 9
train_from = model_common1_epoch8_17.79.t7
max_batch_size = 80
optim = sgd
learning_rate = 0.5
end_epoch = 15
start_decay_at = 11
layers = 4
brnn = true
rnn_size = 1024
gpuid = 1
curriculum = 1 
save_every = 5000
pre_word_vecs_enc = data_common/train/vec_src-embeddings-200.t7
pre_word_vecs_dec = data_common/train/vec_tgt-embeddings-200.t7
```
**Training detail steps:**
when the epoch 9 training finished, I interrupted the training(March 7), after testing the model, I upgraded openNMT to last version 
```
git pull
luarocks remove opennmt
luarocks make rocks/opennmt-scm-1.rockspec
```
I continue to train epoch10, when training finished，I found validation perplexity value is incorrect.

**Comment by guillaumekln** [comment]
Did you work around it?
